supply/lib/supply/setup.rb

```diff
@@ -40,28 +39,44 @@ def store_metadata(listing)
       end
     end
 
-    def download_images(listing)
+    def download_images(listing, image_format = "png", remove_alpha = false)
       UI.message("🖼️  Downloading images (#{listing.language})")
 
-      # We cannot download existing screenshots as they are compressed
-      # But we can at least download the images
       require 'net/http'
 
-      IMAGES_TYPES.each do |image_type|
-        if ['featureGraphic'].include?(image_type)
-          # we don't get all files in full resolution :(
-          UI.message("📵  Due to a limitation of the Google Play API, there is no way for `supply` to download your existing feature graphic. Please copy your feature graphic to `metadata/android/#{listing.language}/images/featureGraphic.png`")
-          next
-        end
+      allowed_imagetypes = [Supply::IMAGES_TYPES, Supply::SCREENSHOT_TYPES].flatten
 
+      allowed_imagetypes.each do |image_type|
         begin
           UI.message("Downloading `#{image_type}` for #{listing.language}...")
 
-          url = client.fetch_images(image_type: image_type, language: listing.language).last
-          next unless url
+          urls = client.fetch_images(image_type: image_type, language: listing.language)
+          next if urls.nil? || urls.empty?
+
+          image_counter = 1 # Used to prefix the downloaded files, so order is preserved.
+          urls.each do |url|
+            url = "#{url}=w6000-h6000" # using huge dimentions intentionally so Google returns the images in their original size they were uploaded
+            path = File.join(metadata_path, listing.language, IMAGES_FOLDER_NAME, image_type, "#{image_counter}_#{listing.language}.#{image_format}")
+            path = File.join(metadata_path, listing.language, IMAGES_FOLDER_NAME, "#{image_type}.#{image_format}") if IMAGES_TYPES.include?(image_type)
+
+            p = Pathname.new(path)
+            FileUtils.mkdir_p(p.dirname.to_s)
+
+            File.write(path, Net::HTTP.get(URI.parse(url))) # Write to tmp `path`
+
+            image = MiniMagick::Image.open(path)
+            image_details = JSON.parse(image.details.to_json, symbolize_names: true)
+
+            is_alpha_present = image_details.keys.any? { |k| k == :Alpha }
+            image.alpha('remove') if is_alpha_present && remove_alpha
+
+            image.format(image_format) # Properly formatting the final image. Sometimes "jpg" or "webp" or "png" formats are returned by Google.
+            image.write(path) # Properly formatted file is written to disk
+            UI.message("\tDownloaded  - #{path}")
```

Comment:
clarify code:
```suggestion

            UI.message("\tDownloaded  - #{path}")
```

Reply:
Is it some trick question? :D What changed here?

Reply:
Added an empty line so the logic is separated from the UI output ;)